### DISCUSSION TITLE
How to obtain the unique misorientation axis?

### DISCUSSION THREAD

**Discussion by SingPr** [mtex-toolbox/mtex#1644] [Ask Anything]
Hi, Ralf. and mtex experts,
       I have the following troubles about how to get the unique axis of orientation difference:
   1. I got the misorientation angle/axis using the following codes:
        ......
```
       ang12 = angle(ori1, ori2,'antipodal') ./ degree
        axis12 = axis(inv(ori1)*ori2,cs)
        round(axis12)
```
There's no doubt that the angle is the min. angle. However, is this the axis that corresponds to the misorientation angle? Or just the crystalline family? How do I write my codes to get the unique misorientation axis corresponding to the min. misorientation angle?

  2. The problem is to calculate misorientation at Subgrain Boundaries. My codes are:
      .......
```
     mori = grains(grain_num).innerBoundary;
   % Compute the density distribution of misorientation axes
    density = calcDensity (mori.misorientation.axis,'halfwidth',3*degree);
   % % plot them
   plot(density,'figSize','small')
   mtexColorbar
 % find the two prefered misorientation axes
    [~,hkl] = max(density,'numLocal',2); round(hkl)
```
   ......
   So, the outputs are the misorientation axes. This is his family too, is it correct? But I want to get the unique misorientation axis (and plot it) instead of his family. Therefore, how should I write my codes?

  Thanks all,
   Sing

**Comment by kilir** [comment]
Hi Sing,
wrt1: yes it is 
```
mori = inv(ori1)*ori2;
angle(mori)
axis(mori)
```

wrt2: yes, you compute a density distribution of all the misorientation axes which correspond to the minimum misorientation angle.

Hope this helps.

Rüdiger

**Reply by SingPr** [reply]
Hi, Rüdiger
  Thank you for your helpful replies.
So, for question 2, you mean that what I calculated is the misorientation axis [hkl] corresponding to the minimum angle, not his family <hkl>, right? As follows:
the output is [hkl], for example: [ -3 10 10], and the plot is：
![untitled](https://user-images.githubusercontent.com/89964907/201339027-8175d378-4520-4281-b830-c6213e3bf3b7.jpg)
But this is a plot after the symmetric operation. Could that be incorrect? In addition, if [hkl] is [-101], it is difficult to represent it in the figure above without symmetric operations. So, my calculations [hkl] and the plot like above are still right?

Thanks a lot
Sing

**Reply by kilir** [reply]
You compute [uvw] corresponding to the minimum angle.
I'm not entirely sure what you mean by "his familiy", but since the the axis obeys crystal symmetry, you can't differentiate between symmetrically equivalent directions [uvw]. You can specify one unique axis in specimen coordinates (`axis(o1,o1)`), see here https://mtex-toolbox.github.io/TiltAndTwistBoundaries.html  at "The misorientation axis in specimen coordinates".
Cheers,
Rüdiger

**Reply by SingPr** [reply]
Hi, Rüdiger
      I may have made some commonsense mistakes.
      “its family": for example, <hkl>---[hkl], [-hkl], [h-kl] and so on.
      The plot example presented above is in a cubic crystal BCC. So, the Miller index at the maximum density distribution of the misorientation axis is [-3 10 10], rather than <-3 10 10>, right?  It may not be meaningful for my research to specify the unique axis in specimen coordinates.
    Thanks
     Sing
